GitUI/GitUICommands.cs

```diff
@@ -1609,13 +1609,35 @@ private bool RunBrowseCommand(IReadOnlyList<string> args)
                 return StartBrowseDialog(null, GetParameterOrEmptyStringAsDefault(args, "-filter"));
             }
 
-            if (Module.TryResolvePartialCommitId(arg, out var objectId))
+            if (tryGetObjectIds(arg, Module, out var objectIds))
             {
-                return StartBrowseDialog(null, GetParameterOrEmptyStringAsDefault(args, "-filter"), objectId);
+                return StartBrowseDialog(null, GetParameterOrEmptyStringAsDefault(args, "-filter"), objectIds.ToArray());
             }
 
             Console.Error.WriteLine($"No commit found matching: {arg}");
             return false;
+
+            static bool tryGetObjectIds(string arg, GitModule module, out List<ObjectId> objectIds)
```

Comment:
We unnecessary allocate memory here - we expect maximum of two items, yet we allocate a dynamic list, and then re-allocate the list into an array.
Allocate and return `ObjectId[2]`